app/models/word.rb

```diff
@@ -1,7 +1,45 @@
 class Word
   include Mongoid::Document
-  field :word_name, type: String
-  field :description, type: String
-  field :syntactical_category, type: String
-  field :semantical-categories, type: String
+
+  has_many :categories
+
+  field :name_de, type: String
+  field :name_en, type: String
+  field :description_de, type: String
+  field :description_en, type: String
+  field :syntactical_category_de, type: String
+  field :syntactical_category_en, type: String
+  field :semantical_categories_de, type: String, :default => nil
+  field :semantical_categories_en, type: Array
+
+
+  # OPTIMIZE: if the word allready exist, it should ask the user if he wants to edit the existing word.
+  validates :name_de, uniqueness: {message: "Dieses Wort gibt es bereits." }
+
+  validates :name_de, presence: true;
+  validates :description_de, presence: true;
+  validates :syntactical_category_de, presence: true;
+  validates :semantical_categories_de, presence: true;
+
+
+
+  def self.get_previous_word(current_word)
```

Comment:
Where do you need those functions? This could get a bit slow if you have several thousands of words.